delira/training/tf_trainer.py

```diff
@@ -60,55 +85,52 @@ def __init__(self, network, save_path,
+        logging_kwargs : dict
+            dictionary containing all logging keyword arguments
         fold : int
             current cross validation fold (0 per default)
         callbacks : list
             initial callbacks to register
         start_epoch : int
             epoch to start training at
+        metric_keys : dict
+            dict specifying which batch_dict entry to use for which metric as 
+            target; default: None, which will result in key "label" for all 
+            metrics
+        convert_batch_to_npy_fn : type, optional
+            function converting a batch-tensor to numpy, per default this is
+            the identity function
+        val_freq : int
+            validation frequency specifying how often to validate the trained
+            model (a value of 1 denotes validating every epoch,
+            a value of 2 denotes validating every second epoch etc.);
+            defaults to 1
         **kwargs :
-            additional keyword arguments
+            Additional keyword arguments
 
         """
 
-        super().__init__(fold, callbacks)
-
-        self.save_path = save_path
-        if os.path.isdir(save_path):
-            logger.warning(
-                "Save Path already exists. Saved Models may be overwritten")
-        else:
-            os.makedirs(save_path)
-
-        # remove prior Trixihandlers and ensure logging of training results to self.save_path
-        # This facilitates visualization of multiple splits/fold inside one tensorboard-instance by means of
-        # different tf.Summary.FileWriters()
-        root_logger = logging.getLogger()
-        for handler in root_logger.handlers:
-            handler.close()
-        root_logger.handlers = []
-        logging.basicConfig(level=logging.INFO,
-                            handlers=[TrixiHandler(TensorboardXLogger, 0, self.save_path)])
-
-        self.losses = losses
-
-        self.metrics = metrics
-
-        self.save_freq = save_freq
-
-        # Whether or not to stop the training
-        # Used for early stopping
-        self.stop_training = False
-        self.start_epoch = start_epoch
+        super().__init__(
+            network, save_path, losses, optimizer_cls, optimizer_params,
+            train_metrics, val_metrics, lr_scheduler_cls, lr_scheduler_params,
+            gpu_ids, save_freq, optim_fn, key_mapping,logging_type,
+            logging_kwargs, fold, callbacks, start_epoch, metric_keys,
+            convert_batch_to_npy_fn, val_freq)
 
         self._setup(network, optim_fn, optimizer_cls, optimizer_params,
-                    lr_scheduler_cls, lr_scheduler_params, gpu_ids)
+                    lr_scheduler_cls, lr_scheduler_params,
+                    key_mapping, convert_batch_to_npy_fn, gpu_ids)
 
         for key, val in kwargs.items():
             setattr(self, key, val)
 
     def _setup(self, network, optim_fn, optimizer_cls, optimizer_params,
```

Comment:
It seems like the `PytorchNetworkTrainer` already supports the `resume` function of the `experiment` class and loads the latest checkpoint. Currently, it is a bit misleading if only one trainer supports this functionality. In general, I'm a big fan to have the option to resume the training.

Reply:
we should imo include a switch on wether to restore from best model or latest model

Reply:
I think the resume inside the trainer should only restore the latest model, since we want t continue training here...

Reply:
okay, still we should update `start_epoch` then